org.eclipse.dawnsci.hdf5.test/src/org/eclipse/dawnsci/hdf5/TestPlugins.java

```diff
@@ -0,0 +1,126 @@
+import org.eclipse.january.dataset.DatasetUtils;
+import org.eclipse.january.dataset.FloatDataset;
+import org.eclipse.january.dataset.IndexIterator;
+import org.junit.Test;
+
+import hdf.hdf5lib.H5;
+import hdf.hdf5lib.HDF5Constants;
+import hdf.hdf5lib.exceptions.HDF5Exception;
+
+/**
+ * Test various filter plugins that are built into the hdf.hdf5lib
+ * plugin
+ */
+public class TestPlugins {
+
+	private final static String DATA_PATH = "/data";
+	private final static int DIM_0 = 10;
+	private final static int DIM_1 = 256;
+	private final static int DIM_2 = 512;
+	private final static long[] DATA_SHAPE = new long[] {DIM_0, DIM_1, DIM_2};
+	private final static long[] DATA_MAX_SHAPE = new long[] {HDF5Constants.H5S_UNLIMITED, DIM_1, DIM_2};
+	private final static long[] DATA_CHUNK = new long[] {1, DIM_1, DIM_2};
+
+	@Test
+	public void testZlibFilter() throws NullPointerException, HDF5Exception {
+		testFilter("test-scratch/testFilterZlib.h5", HDF5Constants.H5Z_FILTER_DEFLATE, 1);
+	}
+
+	@Test
+	public void testLZFFilter() throws NullPointerException, HDF5Exception {
+		testFilter("test-scratch/testFilterLZF.h5", 32000);
+	}
+
+	@Test
+	public void testLZ4Filter() throws NullPointerException, HDF5Exception {
+		testFilter("test-scratch/testFilterLZ4.h5", 32004, 1);
+	}
+
+	@Test
+	public void testBSFilter() throws NullPointerException, HDF5Exception {
+		testFilter("test-scratch/testFilterBS.h5", 32008, 0, 2);
+	}
+
+	public void testFilter(String filename, int filterNumber, int... filterParams) throws NullPointerException, HDF5Exception {
+		createFilteredFile(filename, filterNumber, filterParams);
+		checkFilteredFile(filename);
+	}
+
+	private void createFilteredFile(String filename, int filterNumber, int... filterParams) throws NullPointerException, HDF5Exception {
+		File parent = new File(filename).getParentFile();
+		if (!parent.exists()) {
+			parent.mkdir();
+		}
+		long fileID = H5.H5Fcreate(filename, HDF5Constants.H5F_ACC_TRUNC, HDF5Constants.H5P_DEFAULT, HDF5Constants.H5P_DEFAULT);
+		long dsID = H5.H5Screate_simple(DATA_SHAPE.length, DATA_SHAPE, DATA_MAX_SHAPE);
+		long dcpl = H5.H5Pcreate(HDF5Constants.H5P_DATASET_CREATE);
+		H5.H5Pset_chunk(dcpl, DATA_CHUNK.length, DATA_CHUNK);
+
+		H5.H5Pset_filter(dcpl, filterNumber, HDF5Constants.H5Z_FLAG_MANDATORY, filterParams.length, filterParams);
+
+		long dID = H5.H5Dcreate(fileID, DATA_PATH,
+								HDF5Constants.H5T_NATIVE_FLOAT, dsID,
+								HDF5Constants.H5P_DEFAULT, dcpl, HDF5Constants.H5P_DEFAULT);
+
+		Dataset data = DatasetFactory.createRange(DIM_0 * DIM_1 * DIM_2).reshape(DIM_0, DIM_1, DIM_2);
+		H5.H5Dwrite(dID, HDF5Constants.H5T_NATIVE_DOUBLE, HDF5Constants.H5S_ALL, HDF5Constants.H5S_ALL, HDF5Constants.H5P_DEFAULT, DatasetUtils.serializeDataset(data));
+
+		H5.H5Dclose(dID);
+		H5.H5Pclose(dcpl);
+		H5.H5Sclose(dsID);
+		H5.H5Fclose(fileID);
+	}
+
+	private void checkFilteredFile(String filename) throws NullPointerException, HDF5Exception {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the declaration of thrown exception 'java.lang.NullPointerException' which is a runtime exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)